### ISSUE TITLE
Executor::async / quiet_async and cancel scheduled jobs

### ISSUE THREAD

**Issue by v4hn** [taskflow/taskflow#645]
Hi, thank you for this great project. It's definitely worth the praise it gets. :confetti_ball: 

TLDR:

I'm looking for a seemingly intuitively API that does not seem to exist:

```
tf::Executor executor{3};
for(int i= 0; i < 100; ++i)
  executor.silent_async([](){ ... });
std::this_thread::sleep_for(1s);
executor.cancel_all(); // Is there any way to do this?
```

My context:

I parallelize an anytime system right now where every ran task can make a varying number of new tasks available, but there is no clear static structure between the tasks that I intend to model as a TaskFlow. Instead I use `Executor::quiet_async` from the outside to schedule task execution for each independent task at the moment and that works well so far.

Eventually though, I would like to *stop* processing all scheduled tasks which are not running on a worker yet after some condition (either runtime or a predicate on the computed values) holds.
Is there some recommended mechanism to do that? `Executor.wait_for_all()` would still run all remaining tasks and adding a custom `atomic<bool>` inside the jobs to check before each task would still run through the entire list for no reason.

I could probably model the new tasks recursively with subflows and use `tf::Future::cancel()`, but I'm not convinced that subflows are meant for this as the whole nature of the jobs are iterative increments on a structure and taskflow builds up some information on the nesting structure of subflows.

Using `Executor::run_until` would unnecessarily synchronize between repetitions of a static workflow though, when I really want to check asynchronously after each task finished whether a new one should still be started.

Is this use case not intended for taskflow, do I have a wrong angle on this, or am I just too blind to find the right API here?

**Comment by tsung-wei-huang** [comment]
Hi @v4hn , currently, cancellation is only supported on a per-taskflow basis. You cannot cancel all scheduled async tasks but only submitted taskflows.

**Comment by v4hn** [comment]
Thank you for the clarification.
Assuming this falls into the intended user-cases of the framework, please consider it as a future API extension. :bouquet:

**Comment by tsung-wei-huang** [comment]
@v4hn Will consider it in the future extension! Thank you :)
